src/input-field.tsx

```diff
@@ -113,19 +115,20 @@ export const InputField = React.forwardRef<HTMLInputElement | HTMLTextAreaElemen
         } else if (type === 'checkbox') {
             checked = !!field.value
         }
+        const value = props.value || field.value || '';
         const input = (
             <InputComponent
                 {...field}
                 {...props}
-                value={props.value || field.value || ''}
+                value={value}
                 id={id}
                 ref={ref}
                 checked={checked}
                 onChange={onChange}
                 disabled={Boolean(readOnly || propsDisabled)}
                 onBlur={onBlur}
                 readOnly={readOnly}
-                placeholder={label == null ? 'placeholder' : label}
+                placeholder={placeholder}
```

Comment:
we'll need to make sure we always render a placeholder of some sort, otherwise the select looks weird, could probably just do `placeholder={placeholder || 'placeholder'}`

https://getbootstrap.com/docs/5.1/forms/floating-labels/   I also found how they do it: https://github.com/twbs/bootstrap/blob/main/scss/forms/_floating-labels.scss#L43-L51